src/crypto/crypto_util.cc

```diff
@@ -236,7 +236,12 @@ ByteSource& ByteSource::operator=(ByteSource&& other) noexcept {
 }
 
 std::unique_ptr<BackingStore> ByteSource::ReleaseToBackingStore() {
-  CHECK_NOT_NULL(allocated_data_);
+  CHECK_IMPLIES(size_ == 0, allocated_data_ == nullptr);
+  if (size_ == 0) {
+    return ArrayBuffer::NewBackingStore(
+        nullptr, 0, [](void* data, size_t length, void* deleter_data) {},
+        nullptr);
+  }
```

Comment:
Did this break anything?

Reply:
None of the tests failed locally. We'll see on the broader CI run

Reply:
I mean, I’m just surprised, because I wouldn’t have expected the new assertion here to necessarily be valid under all circumstances. So I’d be curious to know more context :)

Reply:
It appears to hold for everything in our test suite! And really, it *should* hold in general.

Reply:
Ok, but at this risk of being a bit rude: Why *are* we making this change?

Reply:
Specifically, all of our uses of `ByteSource::Allocated()` use `MallocOpenSSL()` which has a corresponding `CHECK_IMPLIES(mem == nullptr, count == 0)`... The `ByteSource::Foreign` variant does not make use of this and `ToArrayBuffer()` (which uses `ReleaseToBackingStore()`) is based on `allocated_data_`.

Reply:
Ok, even more specifically: *Why* do we add another code path when the existing code path would handle `allocated_data_ == nullptr` just fine and with the exact same amount of overhead? It’s only a small bit of extra complexity, but it still seems like entirely avoidable complexity to me?

Reply:
Yeah, we could. The `OPENSSL_clear_free` in the ReleaseToBackingStore should be a non-op if the data is a nullptr...

Reply:
Ok, updated. Removed the extra branch and changed the CHECK_IMPLIES to validate that `allocated_data_` is not nullptr only if size > 0.